src/handlers/autolabel.rs

```diff
@@ -97,16 +85,21 @@ pub(super) async fn parse_input(
 
                 // Treat the following situations as a "new PR":
                 // 1) New PRs opened as non-draft
-                // 2) PRs opened as draft that are marked as "ready for review" for the first time.
+                // 2) PRs opened as draft that are marked as "ready for review".
                 let is_new_non_draft_pr =
                     event.action == IssuesAction::Opened && !event.issue.draft;
-                let is_first_time_ready_for_review = event.action == IssuesAction::ReadyForReview
-                    && !state.data.new_pr_labels_applied;
-                if cfg.new_pr && (is_new_non_draft_pr || is_first_time_ready_for_review) {
+                let is_ready_for_review = event.action == IssuesAction::ReadyForReview;
+                if cfg.new_pr && (is_new_non_draft_pr || is_ready_for_review) {
                     autolabels.push(Label {
                         name: label.to_owned(),
                     });
-                    state.data.new_pr_labels_applied = true;
+                }
+
+                // If a PR is converted to draft remove all the "new PR" labels
+                if cfg.new_pr && event.action == IssuesAction::ConvertedToDraft {
```

Comment:
I wonder if we should also do this when the PR is closed :thinking: There are many closed PRs marked as waiting on review.

Reply:
https://github.com/rust-lang/rust/pulls?q=is%3Apr+label%3AS-waiting-on-review+